src/monmove.cpp

```diff
@@ -683,7 +686,65 @@ void monster::move()
                 // Friendly fire and pushing are always bad choices - they take a lot of time
                 bad_choice = true;
             }
+            //Shove vehicles out of way
+            if( this->has_flag( MF_PUSH_VEH ) ) {
```

Comment:
`monster::move()` is already a behemoth of a function. Please consider extracting a smaller function for the new code.

Reply:
I'm considering making it a special attack.

Reply:
Done.